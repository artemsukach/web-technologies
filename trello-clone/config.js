module.exports = {
    tokenKey: "very-Secret_key"
}
