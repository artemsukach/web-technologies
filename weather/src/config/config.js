import snow from '../img/snow.png';
import wind from '../img/wind.png';
import sun from '../img/sun.png';
import rain from '../img/rain.png';

export const WEATHER = [
  {
    date: '02/28/2022',
    minTemperature: -3,
    maxTemperature: +4,
    atmosphericPressure: 720,
    relativeHumidity: 80,
    windSpeed: 8,
    windDirection: 'South',
    description: snow,
  },
  {
    date: '03/01/2022',
    minTemperature: -2,
    maxTemperature: +6,
    atmosphericPressure: 735,
    relativeHumidity: 77,
    windSpeed: 3.9,
    windDirection: 'South-East',
    description: snow,
  },
  {
    date: '03/02/2022',
    minTemperature: -2,
    maxTemperature: +5,
    atmosphericPressure: 740,
    relativeHumidity: 63,
    windSpeed: 4.3,
    windDirection: 'North',
    description: wind,
  },
  {
    date: '03/03/2022',
    minTemperature: -1,
    maxTemperature: +3,
    atmosphericPressure: 730,
    relativeHumidity: 73,
    windSpeed: 3.4,
    windDirection: 'North-West',
    description: rain,
  },
  {
    date: '03/04/2022',
    minTemperature: 0,
    maxTemperature: +6,
    atmosphericPressure: 729,
    relativeHumidity: 87,
    windSpeed: 3.0,
    windDirection: 'East',
    description: wind,
  },
  {
    date: '03/05/2022',
    minTemperature: 1,
    maxTemperature: 8,
    atmosphericPressure: 738,
    relativeHumidity: 63,
    windSpeed: 3.3,
    windDirection: 'West',
    description: sun,
  },
  {
    date: '03/06/2022',
    minTemperature: 2,
    maxTemperature: 9,
    atmosphericPressure: 742,
    relativeHumidity: 78,
    windSpeed: 3.4,
    windDirection: 'South-West',
    description: sun,
  },
];

export const DAYS = [
  'Sunday',
  'Monday',
  'Tuesday',
  'Wednesday',
  'Thursday',
  'Friday',
  'Saturday',
];

export const MONTHS = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
];
