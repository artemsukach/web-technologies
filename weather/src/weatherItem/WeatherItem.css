.weather__item {
  border-radius: 10px;
  background-color: #e0e0e8;
  text-align: center;
}

.weather__item+.weather__item{
  margin-left: 10px;
}