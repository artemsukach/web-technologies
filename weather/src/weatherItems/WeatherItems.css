.weather__title{
  text-align: center;
}

.weather__items{
  display: flex;
  max-width: 1170px;
  margin: 0 auto;
  justify-content: space-between;
}

