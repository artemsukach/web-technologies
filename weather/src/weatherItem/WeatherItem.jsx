import React from 'react';
import './WeatherItem.css';
import PropTypes from 'prop-types';
import { transformDate } from '../utils/transformDate';
import { getMonthNum } from '../utils/getMonthNum';
import { getDateNum } from '../utils/getDateNum';
import { getDayNum } from '../utils/getDayNum';
import { getYearNum } from '../utils/getYearNum';
import { WeatherDay } from '../weatherItemComponent/WeatherDay.jsx';
import { WeatherDate } from '../weatherItemComponent/WeatherDate.jsx';
import { WeatherMonth } from '../weatherItemComponent/WeatherMonth.jsx';
import { WeatherYear } from '../weatherItemComponent/WeatherYear.jsx';
import { WeatherDescription } from '../weatherItemComponent/WeatherDescription.jsx';
import { WeatherTemperature } from '../weatherItemComponent/WeatherTemperature.jsx';
import { WeatherPressure } from '../weatherItemComponent/WeatherPressure.jsx';
import { WeatherHumidity } from '../weatherItemComponent/WeatherHumidity.jsx';
import { WeatherWindSpeed } from '../weatherItemComponent/WeatherWindSpeed.jsx';
import { WeatherWindDirection } from '../weatherItemComponent/WeatherWindDirection.jsx';

function WeatherItem({
  date,
  minTemperature,
  maxTemperature,
  atmosphericPressure,
  relativeHumidity,
  windSpeed,
  windDirection,
  description,
}) {
  const newDate = new Date(date);
  const { day, dateNum, month, year } = transformDate(
    newDate,
    getMonthNum,
    getDateNum,
    getDayNum,
    getYearNum
  );

  return (
    <div className="weather__item">
      <WeatherDay day={day} />
      <WeatherDate dayNum={dateNum} />
      <WeatherMonth month={month} />
      <WeatherYear year={year} />
      <WeatherDescription description={description} />
      <WeatherTemperature
        minTemperature={minTemperature}
        maxTemperature={maxTemperature}
      />
      <WeatherPressure atmosphericPressure={atmosphericPressure} />
      <WeatherHumidity relativeHumidity={relativeHumidity} />
      <WeatherWindSpeed windSpeed={windSpeed} />
      <WeatherWindDirection windDirection={windDirection} />
    </div>
  );
}

WeatherItem.propTypes = {
  date: PropTypes.string,
  minTemperature: PropTypes.number,
  maxTemperature: PropTypes.number,
  atmosphericPressure: PropTypes.number,
  relativeHumidity: PropTypes.number,
  windSpeed: PropTypes.number,
  windDirection: PropTypes.string,
  description: PropTypes.string,
}

export default WeatherItem;
