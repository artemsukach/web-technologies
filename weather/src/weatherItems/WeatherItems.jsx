import './WeatherItems.css';
import React from 'react';
import WeatherItem from '../weatherItem/WeatherItem.jsx';
import PropTypes from 'prop-types';

function WeatherItems({ weather }) {
  return (
    <div className="weather">
      <h1 className="weather__title">Weather</h1>
      <div className="weather__items">
        {weather.map((item) => (
          <WeatherItem key={item.date} {...item} />
        ))}
      </div>
    </div>
  );
}

WeatherItems.propTypes = {
  weather: PropTypes.array.isRequired
}

WeatherItems.defaultProps = {
  weather: []
}

export default WeatherItems;
