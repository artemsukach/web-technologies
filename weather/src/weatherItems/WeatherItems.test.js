import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import WeatherItems from './WeatherItems';
import { WEATHER } from '../config/config.js';
import WeatherItem from '../weatherItem/WeatherItem';

Enzyme.configure({ adapter: new Adapter() });

describe('WeatherItems', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<WeatherItems weather={WEATHER} />);
  });

  test('renders weather wrapper', () => {
    expect(wrapper.find('.weather').length).toBe(1);
  });

  test('shoud contain h1', () => {
    expect(wrapper.find('.weather__title').length).toBe(1);
    expect(wrapper.find('.weather__title').text()).toBe('Weather');
  });

  test('shoud contain weather items wrapper', () => {
    expect(wrapper.find('.weather__items').length).toBe(1);
  });

  test('shoud contain weather item component', () => {
    expect(wrapper.find(WeatherItem).length).toBe(WEATHER.length);
  });

  test('shoud match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
