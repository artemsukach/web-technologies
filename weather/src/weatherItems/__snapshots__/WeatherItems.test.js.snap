// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherItems shoud match snapshot 1`] = `"<div class=\\"weather\\"><h1 class=\\"weather__title\\">Weather</h1><div class=\\"weather__items\\"><div class=\\"weather__item\\"><p class=\\"weather__day\\">Monday</p><p class=\\"weather__date\\">28</p><p class=\\"weather__month\\">February</p><p class=\\"weather__year\\">2022 year</p><img class=\\"weather__description\\" src=\\"snow.png\\" alt=\\"weather\\"/><div class=\\"weather__temperature-wrapper\\"><div class=\\"weather__min-temperature\\"><p class=\\"weather__min-description\\">min</p><p class=\\"weather__min-value\\">-3<span>°</span></p></div><div class=\\"weather__max-temperature\\"><p class=\\"weather__max-description\\">max</p><p class=\\"weather__max-value\\">4<span>°</span></p></div></div><p class=\\"weather__atmospheric-pressure\\">Atmospheric pressure: 720</p><p class=\\"weather__humidity\\">Reletive humidity: 80</p><p class=\\"weather__wind-speed\\">Wind speed: 8</p><p class=\\"weather__wind-direction\\">Wind direction: South</p></div><div class=\\"weather__item\\"><p class=\\"weather__day\\">Tuesday</p><p class=\\"weather__date\\">1</p><p class=\\"weather__month\\">March</p><p class=\\"weather__year\\">2022 year</p><img class=\\"weather__description\\" src=\\"snow.png\\" alt=\\"weather\\"/><div class=\\"weather__temperature-wrapper\\"><div class=\\"weather__min-temperature\\"><p class=\\"weather__min-description\\">min</p><p class=\\"weather__min-value\\">-2<span>°</span></p></div><div class=\\"weather__max-temperature\\"><p class=\\"weather__max-description\\">max</p><p class=\\"weather__max-value\\">6<span>°</span></p></div></div><p class=\\"weather__atmospheric-pressure\\">Atmospheric pressure: 735</p><p class=\\"weather__humidity\\">Reletive humidity: 77</p><p class=\\"weather__wind-speed\\">Wind speed: 3.9</p><p class=\\"weather__wind-direction\\">Wind direction: South-East</p></div><div class=\\"weather__item\\"><p class=\\"weather__day\\">Wednesday</p><p class=\\"weather__date\\">2</p><p class=\\"weather__month\\">March</p><p class=\\"weather__year\\">2022 year</p><img class=\\"weather__description\\" src=\\"wind.png\\" alt=\\"weather\\"/><div class=\\"weather__temperature-wrapper\\"><div class=\\"weather__min-temperature\\"><p class=\\"weather__min-description\\">min</p><p class=\\"weather__min-value\\">-2<span>°</span></p></div><div class=\\"weather__max-temperature\\"><p class=\\"weather__max-description\\">max</p><p class=\\"weather__max-value\\">5<span>°</span></p></div></div><p class=\\"weather__atmospheric-pressure\\">Atmospheric pressure: 740</p><p class=\\"weather__humidity\\">Reletive humidity: 63</p><p class=\\"weather__wind-speed\\">Wind speed: 4.3</p><p class=\\"weather__wind-direction\\">Wind direction: North</p></div><div class=\\"weather__item\\"><p class=\\"weather__day\\">Thursday</p><p class=\\"weather__date\\">3</p><p class=\\"weather__month\\">March</p><p class=\\"weather__year\\">2022 year</p><img class=\\"weather__description\\" src=\\"rain.png\\" alt=\\"weather\\"/><div class=\\"weather__temperature-wrapper\\"><div class=\\"weather__min-temperature\\"><p class=\\"weather__min-description\\">min</p><p class=\\"weather__min-value\\">-1<span>°</span></p></div><div class=\\"weather__max-temperature\\"><p class=\\"weather__max-description\\">max</p><p class=\\"weather__max-value\\">3<span>°</span></p></div></div><p class=\\"weather__atmospheric-pressure\\">Atmospheric pressure: 730</p><p class=\\"weather__humidity\\">Reletive humidity: 73</p><p class=\\"weather__wind-speed\\">Wind speed: 3.4</p><p class=\\"weather__wind-direction\\">Wind direction: North-West</p></div><div class=\\"weather__item\\"><p class=\\"weather__day\\">Friday</p><p class=\\"weather__date\\">4</p><p class=\\"weather__month\\">March</p><p class=\\"weather__year\\">2022 year</p><img class=\\"weather__description\\" src=\\"wind.png\\" alt=\\"weather\\"/><div class=\\"weather__temperature-wrapper\\"><div class=\\"weather__min-temperature\\"><p class=\\"weather__min-description\\">min</p><p class=\\"weather__min-value\\">0<span>°</span></p></div><div class=\\"weather__max-temperature\\"><p class=\\"weather__max-description\\">max</p><p class=\\"weather__max-value\\">6<span>°</span></p></div></div><p class=\\"weather__atmospheric-pressure\\">Atmospheric pressure: 729</p><p class=\\"weather__humidity\\">Reletive humidity: 87</p><p class=\\"weather__wind-speed\\">Wind speed: 3</p><p class=\\"weather__wind-direction\\">Wind direction: East</p></div><div class=\\"weather__item\\"><p class=\\"weather__day\\">Saturday</p><p class=\\"weather__date\\">5</p><p class=\\"weather__month\\">March</p><p class=\\"weather__year\\">2022 year</p><img class=\\"weather__description\\" src=\\"sun.png\\" alt=\\"weather\\"/><div class=\\"weather__temperature-wrapper\\"><div class=\\"weather__min-temperature\\"><p class=\\"weather__min-description\\">min</p><p class=\\"weather__min-value\\">1<span>°</span></p></div><div class=\\"weather__max-temperature\\"><p class=\\"weather__max-description\\">max</p><p class=\\"weather__max-value\\">8<span>°</span></p></div></div><p class=\\"weather__atmospheric-pressure\\">Atmospheric pressure: 738</p><p class=\\"weather__humidity\\">Reletive humidity: 63</p><p class=\\"weather__wind-speed\\">Wind speed: 3.3</p><p class=\\"weather__wind-direction\\">Wind direction: West</p></div><div class=\\"weather__item\\"><p class=\\"weather__day\\">Sunday</p><p class=\\"weather__date\\">6</p><p class=\\"weather__month\\">March</p><p class=\\"weather__year\\">2022 year</p><img class=\\"weather__description\\" src=\\"sun.png\\" alt=\\"weather\\"/><div class=\\"weather__temperature-wrapper\\"><div class=\\"weather__min-temperature\\"><p class=\\"weather__min-description\\">min</p><p class=\\"weather__min-value\\">2<span>°</span></p></div><div class=\\"weather__max-temperature\\"><p class=\\"weather__max-description\\">max</p><p class=\\"weather__max-value\\">9<span>°</span></p></div></div><p class=\\"weather__atmospheric-pressure\\">Atmospheric pressure: 742</p><p class=\\"weather__humidity\\">Reletive humidity: 78</p><p class=\\"weather__wind-speed\\">Wind speed: 3.4</p><p class=\\"weather__wind-direction\\">Wind direction: South-West</p></div></div></div>"`;
